ui-admin/src/portal/images/SiteImageUploadModal.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { ChangeEvent, useRef, useState } from 'react'
+import { Modal } from 'react-bootstrap'
+import Api, { SiteImageMetadata } from 'api/api'
+import { doApiLoad } from 'api/api-utils'
+import LoadingSpinner from 'util/LoadingSpinner'
+import { successNotification } from 'util/notifications'
+import { Store } from 'react-notifications-component'
+import { Button } from 'components/forms/Button'
+import { LoadedPortalContextT } from '../PortalProvider'
+
+/** Renders a modal for an admin to submit a sample collection kit request. */
+export default function SiteImageUploadModal({
+  portalContext,
+  onDismiss, onSubmit,
+  existingImage
+}: {
+    portalContext: LoadedPortalContextT,
+    onDismiss: () => void,
+    existingImage?: SiteImageMetadata,
+    onSubmit: () => void }) {
+  const [isLoading, setIsLoading] = useState(false)
+
+  const [fileName, setFileName] = useState(existingImage?.cleanFileName ?? '')
+
+  const handleFileChange = (newFile: File) => {
+    // only autofill the name field if this is a new image
+    if (!existingImage) {
+      setFileName(cleanFileName(newFile.name))
+    }
+  }
+
+  const { file, FileChooser } = useFileUploadButton(handleFileChange)
+  const uploadImage = () => {
+    if (!file) { return }
+    const version = existingImage?.version ? existingImage.version + 1 : 1
```

Comment:
```suggestion
  const uploadImage = () => {
    if (!file) { return }
    
    if (!file.type.match(/image\/(gif|ico|jpeg|jpg|png|svg|webp)/)) {
      Store.addNotification(failureNotification('File must be an image.'))
      return
    }

    if (file.size > 10 * 1024 * 1024) {
      Store.addNotification(failureNotification('File size exceeds the 10MB limit.'))
      return
    }
    
    const version = existingImage?.version ? existingImage.version + 1 : 1
```

Quick pass at adding some frontend validation for file type and file size

Reply:
nice, I'm going to go ahead and put this logic in the component itself so you can't submit